Source/carthage/Outdated.swift

```diff
@@ -54,13 +55,40 @@ public struct OutdatedCommand: CommandProtocol {
 
 				if !outdatedDependencies.isEmpty {
 					carthage.println(formatting.path("The following dependencies are outdated:"))
+
 					for (project, current, updated, latest) in outdatedDependencies {
+						let versionColor: Color.Named.Color
+						switch (current, updated, latest) {
+						case (_, updated, latest) where updated == latest:
+							// Update available and applicable
+							versionColor = .green
+						case (current, updated, latest) where current != updated && updated != latest:
+							// Update availabe and applicable, but not to the latest version due to version lock
+							versionColor = .yellow
+						case (current, updated, latest) where current == updated:
+							// Update available, but not applicable due to version lock
+							versionColor = .red
+						default:
+							versionColor = .white
```

Comment:
Not sure, is this actually down a dead codepath, considering the logic of `outdatedDependencies(_:useNewResolver:resolver:)`?

If actually in the codepath, using `.white` here can’t function as a workaround, considering [Terminal.app](https://www.apple.com/search/terminal.app)’s default theme as of recent years has a white background.

Subsequently, if we actually need to display that text, something worth considering is to separate the formatting from the logic — an enum declared in `Outdated.swift` that has formatting and the legend for colors in `Formatting.swift`:

```swift
enum UpdateAvailabilityAndApplicability: String {
	case updateToLatestAvailableAndApplicable =
		"Will be updated to the newest version."
	case updateAvailableYetLatestUnapplicableDueToVersionSpecifier =
		"Will be updated, but not to the newest version."
	case everyAvailableUpdateUnapplicableDueToVersionSpecifier =
		"Will not be updated because of the specified version in Cartfile."
	case noAvailableUpdates =
		"Will not be updated: no updates available."
}

/* in Formatting.swift */
extension UpdateAvailabilityAndApplicability {
	fileprivate static let dict: [Self: Color.Named.Color] = [
		updateToLatestAvailableAndApplicable: .green,
		updateAvailableYetLatestUnapplicableDueToVersionSpecifier: .yellow,
		everyAvailableUpdateUnapplicableDueToVersionSpecifier: .red
	]
}

extension Formatting {
	public subscript(_ index: UpdateAvailabilityAndApplicability) -> Wrap {
		guard self.isColorful, let color = UpdateAvailabilityAndApplicability.dict[index] else {
			return { $0 }
		}

		return Color.Wrap(foreground: color).wrap
	}

	public var legendForOutdatedCommand: String? {
		guard self.isColorful else { return nil }

		let header = "Legend — <color>\t• «what happens when you run `carthage update`»:\n"
	
		return UpdateAvailabilityAndApplicability.dict.reduce(into: header) {
			let (situation, color) = $1

			let tabs = String(
				repeating: "\t",
				count: color == .yellow || color == .magenta ? 1 : 2
			)

			let colorDescription = Color.Wrap(foreground: color)
				.wrap("<" + String(describing: color) + ">")

			$0.append(colorDescription + tabs + "• " + situation.rawValue + "\n")
		}
	}
}

/* back in Outdated.swift’s `run` for-loop */
let style = formatting[updateAvailabilityAndApplicability]
style(current.description) + " -> " + style(updated.description) + " (Latest: \(latest))"
```

Reply:
Yeah, right now this is not reachable due to the logic of `outdatedDependencies(_:useNewResolver:resolver:)` which will not include dependencies that that are **not** outdated (where no update is available). But that can not be guaranteed and I like your suggestion anyway.